kaldb/src/main/java/com/slack/kaldb/metadata/service/ServicePartitionMetadata.java

```diff
@@ -85,4 +87,29 @@ public static Metadata.ServicePartitionMetadata toServicePartitionMetadataProto(
         .addAllPartitions(metadata.partitions)
         .build();
   }
+
+  /**
+   * get partitions that match on two criteria 1. index name 2. partitions that have an overlap with
+   * the query window
+   */
+  public static List<ServicePartitionMetadata> findPartitionsToQuery(
+      ServiceMetadataStore serviceMetadataStore,
+      long startTimeEpochMs,
+      long endTimeEpochMs,
+      String indexName) {
+    return serviceMetadataStore
+        .getCached()
+        .stream()
+        .filter(serviceMetadata -> serviceMetadata.name.equals(indexName))
+        .flatMap(
+            serviceMetadata -> serviceMetadata.partitionConfigs.stream()) // will always return one
+        .filter(
+            partitionMetadata ->
+                containsDataInTimeRange(
```

Comment:
Instead of importing this as a static, just call it with `ChunkInfo.containsDataInTimeRange()`